### PR TITLE
Refactored Presenters, Tests, URLs, and Views

### PR DESCRIPTION
**Pull request by clintonb**
All of these files were growing quite large. This refactoring splits them by major component--enrollment, engagement, performance. The resulting files are smaller and more maintainable.

@rocha @dsjen

### CODE REVIEW

File: analytics_dashboard/courses/presenters/performance.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+from collections import namedtuple
+
+from analyticsclient.exceptions import NotFoundError
+
+import courses.utils as utils
+
+# Stores the answer distribution return from CoursePerformancePresenter
```
Comment (dsjen) [inline analytics_dashboard/courses/presenters/performance.py]: This comment belongs with `AnswerDistributionEntry`.

File: analytics_dashboard/courses/tests/test_views/__init__.py [hunk 2]
```diff
@@ -14,12 +13,35 @@
...
+from courses.tests.utils import set_empty_permissions, get_mock_api_enrollment_data, mock_course_name
 
 
 DEMO_COURSE_ID = 'course-v1:edX+DemoX+Demo_2014'
 DEPRECATED_DEMO_COURSE_ID = 'edX/DemoX/Demo_Course'
-COURSE_API_URL = 'http://example.com/api/server/'
+
+
+class CourseAPIMixin(SwitchMixin):
+    COURSE_API_COURSE_LIST = {'results': [
```
Comment (dsjen) [inline analytics_dashboard/courses/tests/test_views/__init__.py:28]: A note about how this is used would be helpful.

File: analytics_dashboard/courses/views/performance.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import logging
+
+from django.conf import settings
+
+from django.utils.translation import ugettext_lazy as _
```
Comment (dsjen) [inline analytics_dashboard/courses/views/performance.py:5]: Minor: the django imports could be collocated.

File: analytics_dashboard/courses/views/__init__.py [hunk 4]
```diff
@@ -0,0 +1,505 @@
+import copy
+import json
+import logging
+import re
+
+from django.core.cache import cache
+from django.utils.functional import cached_property
+from edx_api_client.auth import TokenAuth
```
Comment (dsjen) [inline analytics_dashboard/courses/views/__init__.py]: Can the imports be organized better?

File: analytics_dashboard/courses/views/__init__.py [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+                cache.set(key, info)
+            except HttpClientError as e:
+                logger.error("Unable to retrieve course info for {}: {}".format(course_id, e))
+                info = {}
+
+        return info
+
+    def get_courses(self, course_ids=None):
+        if course_ids:
+            ','.join(course_ids)
```
Comment (dsjen) [inline analytics_dashboard/courses/views/__init__.py]: Is the result of the join supposed to be saved anywhere?

File: analytics_dashboard/courses/views/__init__.py [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+    pass
+
+
+class CourseHome(CourseTemplateView):
+    template_name = 'courses/home.html'
+    page_name = 'course_home'
+    page_title = _('Course Home')
+
+    def get_navbar_items(self):
+        return [
```
Comment (dsjen) [inline analytics_dashboard/courses/views/__init__.py]: `CourseNavBarMixin.get_primary_nav_items` and this method return the same list.  Can the values be combined?

File: analytics_dashboard/courses/presenters/performance.py (hunk 1 above)
Comment (clintonb) [inline analytics_dashboard/courses/presenters/performance.py]: Fixed.

File: analytics_dashboard/courses/tests/test_views/__init__.py (hunk 2 above)
Comment (clintonb) [inline analytics_dashboard/courses/tests/test_views/__init__.py:28]: Added.

File: analytics_dashboard/courses/views/__init__.py (hunk 4 above)
Comment (clintonb) [inline analytics_dashboard/courses/views/__init__.py]: Alphabetized. I am relying on PyCharm to optimize code/imports which does so via the [PEP 8 standard](https://www.python.org/dev/peps/pep-0008/#imports).

File: analytics_dashboard/courses/views/__init__.py (hunk 5 above)
Comment (clintonb) [inline analytics_dashboard/courses/views/__init__.py]: Fixed.

File: analytics_dashboard/courses/views/__init__.py (hunk 6 above)
Comment (clintonb) [inline analytics_dashboard/courses/views/__init__.py]: Merged.

**Comment by dsjen** [comment]
It looks like a test case is failing.  Otherwise, LGTM!

**Pull request merged**